api/src/main/java/com/okta/idx/sdk/api/client/BaseIDXClient.java

```diff
@@ -401,7 +401,7 @@ public IDXResponse recover(RecoverRequest recoverRequest, String href) throws Pr
         try {
             Request request = new DefaultRequest(
                     HttpMethod.POST,
-                    href,
+                    Strings.hasText(href) ? href : clientConfiguration.getBaseUrl() + "/idp/idx/recover",
```

Comment:
We should file a server bug for this.

Reply:
I don't think this is a server side bug. We have to do this here because there might be a need to call `recover` in one of the cases - a) without a `href` (without following any remediation) b) with a `href` (part of remediation option). Case a is being dealt with in this case (recovery password - one step identify).